### PR TITLE
Tracing values

### PR DESCRIPTION
**Pull request by mattrutherford**
Closes https://github.com/paritytech/substrate/issues/6678

Use a discrete map for each type recorded, instead of defaulting to debug / string.

### CODE REVIEW

File: client/tracing/src/lib.rs [hunk 1]
```diff
@@ -78,72 +77,104 @@ pub struct SpanDatum {
...
-	pub values: Visitor,
+	pub values: Values,
 }
 
 /// Holds associated values for a tracing span
-#[derive(Clone, Debug)]
-pub struct Visitor(FxHashMap<String, String>);
+#[derive(Default, Clone, Debug)]
+pub struct Values {
+	pub bool_values: FxHashMap<String, bool>,
```
Comment (kianenigma) [inline client/tracing/src/lib.rs:86]: for consistency pub fields should have doc.

File: client/tracing/src/lib.rs [hunk 2]
```diff
@@ -78,72 +77,104 @@ pub struct SpanDatum {
...
+
+impl Values {
+	/// Returns a new instance of Values
+	pub fn new() -> Self {
+		Values {
+			bool_values: Default::default(),
+			i64_values: Default::default(),
+			u64_values: Default::default(),
+			string_values: Default::default()
+		}
```
Comment (kianenigma) [inline client/tracing/src/lib.rs:100]:
```suggestion
		Default::default()
```

File: client/tracing/src/lib.rs (hunk 1 above)
Comment (mattrutherford) [reply client/tracing/src/lib.rs:86]: thanks, missed that

**Comment by mattrutherford** [comment]
bot merge

**Pull request merged**